tests/components/icloud/test_config_flow.py

```diff
@@ -0,0 +1,289 @@
+        }
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "trusted_device"
+
+
+async def test_import_with_cookie(
+    hass: HomeAssistantType, service_with_cookie: MagicMock
+):
+    """Test import step with presence of a cookie."""
+    flow = init_config_flow(hass)
+
+    # import with username and password
+    result = await flow.async_step_import(
+        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == USERNAME
+    assert result["data"][CONF_USERNAME] == USERNAME
+    assert result["data"][CONF_PASSWORD] == PASSWORD
+    assert result["data"][CONF_ACCOUNT_NAME] == ACCOUNT_NAME_FROM_USERNAME
+    assert result["data"][CONF_MAX_INTERVAL] == DEFAULT_MAX_INTERVAL
+    assert result["data"][CONF_GPS_ACCURACY_THRESHOLD] == DEFAULT_GPS_ACCURACY_THRESHOLD
+
+    # import with all
+    result = await flow.async_step_import(
+        {
+            CONF_USERNAME: USERNAME,
+            CONF_PASSWORD: PASSWORD,
+            CONF_ACCOUNT_NAME: ACCOUNT_NAME,
+            CONF_MAX_INTERVAL: MAX_INTERVAL,
+            CONF_GPS_ACCURACY_THRESHOLD: GPS_ACCURACY_THRESHOLD,
+        }
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == USERNAME
+    assert result["data"][CONF_USERNAME] == USERNAME
+    assert result["data"][CONF_PASSWORD] == PASSWORD
+    assert result["data"][CONF_ACCOUNT_NAME] == ACCOUNT_NAME
+    assert result["data"][CONF_MAX_INTERVAL] == MAX_INTERVAL
+    assert result["data"][CONF_GPS_ACCURACY_THRESHOLD] == GPS_ACCURACY_THRESHOLD
+
+
+async def test_abort_if_already_setup(hass: HomeAssistantType):
+    """Test we abort if the account is already setup."""
+    flow = init_config_flow(hass)
+    MockConfigEntry(
+        domain=DOMAIN, data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+    ).add_to_hass(hass)
+
+    # Should fail, same USERNAME (import)
+    result = await flow.async_step_import(
+        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "username_exists"
+
+    # Should fail, same ACCOUNT_NAME (import)
+    result = await flow.async_step_import(
+        {
+            CONF_USERNAME: "other_username@icloud.com",
+            CONF_PASSWORD: PASSWORD,
+            CONF_ACCOUNT_NAME: ACCOUNT_NAME_FROM_USERNAME,
+        }
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "username_exists"
+
+    # Should fail, same USERNAME (flow)
+    result = await flow.async_step_user(
+        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_USERNAME: "username_exists"}
+
+
+async def test_abort_on_login_failed(hass: HomeAssistantType):
```

Comment:
This test ends with a form result. Should we rename the test or change some logic?

Reply:
Indeed, fixed